File: 1888_Minimum_Number_of_Flips_to_Make_the_Binary_String_Alternating/solution.c
int minFlips(char* s) {
    int n = strlen(s), alt1 = 0, alt2 = 0, res = INT_MAX;
    char* t = malloc(2 * n + 1);
    strcpy(t, s); strcpy(t + n, s);
    for (int i = 0; i < 2 * n; i++) {
        alt1 += t[i] - '0' != i % 2;
        alt2 += t[i] - '0' != (i + 1) % 2;
        if (i >= n) { alt1 -= t[i - n] - '0' != (i - n) % 2; alt2 -= t[i - n] - '0' != (i - n + 1) % 2; }
        if (i >= n - 1) res = fmin(res, fmin(alt1, alt2));
    }
    free(t);
    return res;
}

File: 1888_Minimum_Number_of_Flips_to_Make_the_Binary_String_Alternating/solution.py
class Solution:
    def minFlips(self, s: str) -> int:
        n = len(s)
        s += s
        alt1 = alt2 = res = 0
        res = float("inf")
        for i, c in enumerate(s):
            alt1 += c != str(i % 2)
            alt2 += c != str((i + 1) % 2)
            if i >= n:
                alt1 -= s[i - n] != str((i - n) % 2)
                alt2 -= s[i - n] != str((i - n + 1) % 2)
            if i >= n - 1:
                res = min(res, alt1, alt2)
        return res